test/integration/resourcemanager/highavailabilityconfig/highavailabilityconfig_test.go

```diff
@@ -588,134 +624,232 @@ var _ = Describe("HighAvailabilityConfig tests", func() {
-									},
-									corev1.TopologySpreadConstraint{
-										TopologyKey:       corev1.LabelTopologyZone,
-										MaxSkew:           2,
-										WhenUnsatisfiable: corev1.DoNotSchedule,
-										LabelSelector:     &metav1.LabelSelector{MatchLabels: labels},
-									},
-								))
-							})
-						}
+					Context("when current replica counts are higher than calculated ones", func() {
+						var newMinReplicas, newMaxReplicas int32
 
-						Context("scaled w/ HPA", func() {
-							BeforeEach(func() {
-								hpa.Spec.ScaleTargetRef.Name = getObj().GetName()
-								hpa.Spec.MaxReplicas = int32(2 * len(zones))
-							})
+						BeforeEach(func() {
+							newMinReplicas = 12
+							newMaxReplicas = 16
 
-							test()
+							setMinReplicas(&newMinReplicas)
+							setMaxReplicas(newMaxReplicas)
 						})
 
-						Context("scaled w/ HVPA", func() {
-							BeforeEach(func() {
-								hvpa.Spec.TargetRef.Name = getObj().GetName()
-								hvpa.Spec.Hpa.Template.Spec.MaxReplicas = int32(2 * len(zones))
-							})
-
-							test()
+						It("should not modify replica counts", func() {
+							Expect(getMinReplicas()).To(PointTo(Equal(newMinReplicas)))
+							Expect(getMaxReplicas()).To(Equal(newMaxReplicas))
 						})
 					})
 				})
 			})
-		})
-	}
+		}
 
-	Context("for deployments", func() {
-		BeforeEach(func() {
-			hpa.Spec.ScaleTargetRef.Kind = "Deployment"
-			hvpa.Spec.TargetRef.Kind = "Deployment"
-		})
+		Context("for HVPAs", func() {
+			var hvpa *hvpav1alpha1.Hvpa
 
-		JustBeforeEach(func() {
-			By("Create Deployment")
-			Expect(testClient.Create(ctx, deployment)).To(Succeed())
-		})
+			BeforeEach(func() {
+				hvpa = &hvpav1alpha1.Hvpa{
+					ObjectMeta: objectMeta,
+					Spec: hvpav1alpha1.HvpaSpec{
+						TargetRef: &autoscalingv2beta1.CrossVersionObjectReference{
+							Kind: "someKind",
+							Name: "someName",
+						},
+					},
+				}
 
-		tests(
-			func() client.Object { return deployment },
-			func() *int32 { return deployment.Spec.Replicas },
-			func(replicas *int32) { deployment.Spec.Replicas = replicas },
-			func() corev1.PodSpec { return deployment.Spec.Template.Spec },
-			func(mutate func(spec *corev1.PodSpec)) { mutate(&deployment.Spec.Template.Spec) },
-		)
-	})
+				scalingObject = hvpa
+			})
 
-	Context("for statefulsets", func() {
-		BeforeEach(func() {
-			hpa.Spec.ScaleTargetRef.Kind = "StatefulSet"
-			hvpa.Spec.TargetRef.Kind = "StatefulSet"
+			tests(
+				func() *metav1.ObjectMeta { return &hvpa.ObjectMeta },
+				func() *int32 { return hvpa.Spec.Hpa.Template.Spec.MinReplicas },
+				func(n *int32) { hvpa.Spec.Hpa.Template.Spec.MinReplicas = n },
+				func() int32 { return hvpa.Spec.Hpa.Template.Spec.MaxReplicas },
+				func(n int32) { hvpa.Spec.Hpa.Template.Spec.MaxReplicas = n },
+			)
 		})
 
-		JustBeforeEach(func() {
-			By("Create StatefulSet")
-			Expect(testClient.Create(ctx, statefulSet)).To(Succeed())
-		})
+		Context("for HPAs", func() {
```

Comment:
Can you add an additional test case to also cover the autoscaling `v2beta1` version

Reply:
I tried to do this in the beginning but the `v2beta1` group has been removed in Kubernetes `1.25` [ref](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v125) which is the foundation for our integration tests.

However, I know included `v2beta2` and interestingly enough I found another place to improve, which is setting `MatchPolicy: Equivalent`. See https://github.com/gardener/gardener/blob/423a78306f6a1d16e7dca78dcff925e10a02255d/pkg/operation/botanist/component/resourcemanager/resource_manager.go#L1704